account_id.go

```diff
@@ -151,18 +180,36 @@ func _AccountIDFromProtobuf(accountID *services.AccountID) *AccountID {
 		return nil
 	}
 
-	return &AccountID{
-		Shard:   uint64(accountID.ShardNum),
-		Realm:   uint64(accountID.RealmNum),
-		Account: uint64(accountID.AccountNum),
+	switch t := accountID.Account.(type) {
+	case *services.AccountID_Alias:
+		key, _ := PublicKeyFromBytes(t.Alias)
+		return &AccountID{
+			Shard:    uint64(accountID.ShardNum),
+			Realm:    uint64(accountID.RealmNum),
+			Account:  0,
+			AliasKey: &key,
+		}
+	case *services.AccountID_AccountNum:
+		return &AccountID{
+			Shard:    uint64(accountID.ShardNum),
+			Realm:    uint64(accountID.RealmNum),
+			Account:  uint64(t.AccountNum),
+			AliasKey: nil,
+		}
+	default:
+		return &AccountID{}
 	}
 }
 
 func (id AccountID) _IsZero() bool {
-	return id.Shard == 0 && id.Realm == 0 && id.Account == 0
+	return id.Shard == 0 && id.Realm == 0 && id.Account == 0 && id.AliasKey == nil
 }
 
 func (id AccountID) _Equals(other AccountID) bool {
+	if id.AliasKey != nil && other.AliasKey != nil {
+		return id.Shard == other.Shard && id.Realm == other.Realm && id.Account == other.Account && id.AliasKey.String() == other.AliasKey.String()
```

Comment:
Instead of duplicating the `id.Shard == other.Shard` etc extract `id.AliasKey.String() == other.AliasKey.String()` into a variable and use it in a single return.